recurring_contract/wizards/recurring_invoicer_wizard.py

```diff
@@ -25,12 +37,16 @@ class InvoicerWizard(models.TransientModel):
 
     @api.multi
     def generate(self):
-        date_limit = datetime.date.today() + relativedelta(months=+1)
+
+        @memoize
+        def get_limit_date(advance_months):
+            return datetime.date.today() + relativedelta(months=+advance_months)
 
         recurring_invoicer_obj = self.env['recurring.invoicer']
         contract_groups = self.env['recurring.contract.group'].search([
-            ('next_invoice_date', '<=', date_limit),
             ('next_invoice_date', '!=', False)])
+        contract_groups = contract_groups.filtered(
```

Comment:
Why can't you continue to use the search instead of filtered? The search query would be much faster than the filtered function so I would suggest to revert that.

Reply:
this would require comparing a field to an other one 
```python
(['field_a', '<=', fun(field_b)])
```
I recall already have had to switch to a filter on this type of scenario. Is there an alternative ?

Reply:
Oh I didn't notice you are comparing two fields of the model. You are right that there is no way to do it directly in the search. I'm a bit afraid that we will have a massive performance drop with this change because we are loading almost all groups and then filtering on them. I would be more happy if the search could filter more in front before doing the filter in memory. But I don't have any solution to propose so if you don't find one either, let's move one like that.

Reply:
Maybe the solution would be to execute a SQL query for fetching the ids and then use the browse method. 
```
self.env.cr.execute("""
    SELECT id FROM recurring_contract_group
    WHERE ...
"""")
gids = self.env.cr.fetchall()
contract_goups = self.browse(gids)
```